src/storage/raw/encoded.rs

```diff
@@ -137,13 +130,9 @@ impl<I: Iterator, API: APIVersion> RawEncodeIterator<I, API> {
             }
 
             if *res.as_ref().unwrap() {
-                let raw_value = API::decode_raw_value(self.inner.value())?;
-                if raw_value
-                    .expire_ts
-                    .map(|expire_ts| expire_ts <= self.current_ts)
-                    .unwrap_or(false)
-                {
-                    self.skip_ttl += 1;
+                let raw_value = Api::decode_raw_value(self.inner.value())?;
+                if !raw_value.is_valid(self.current_ts) {
```

Comment:
If it's a delete, should we skip the next keys with same user key?

Reply:
Yes. The underlying `RawMvccIterator` will iterate forward to the next user key.
See [`RawStore::V2`](https://github.com/tikv/tikv/pull/12343/files#diff-a8259649fcd9453b0f9fcb26f241cc7537a19e205657ddf6015f9e6fcd62d3d9R25) & [`RawMvccIterator::next`](https://github.com/tikv/tikv/blob/b60271983288caf7d978f111cd1878b508861327/src/storage/raw/raw_mvcc.rs#L176).